frontend/packages/topology/src/components/dropdowns/ApplicationDropdown.tsx

```diff
@@ -70,16 +74,31 @@ const ApplicationDropdown: React.FC<ApplicationDropdownProps> = ({ namespace, ..
       ...watchedBaseResources[key],
       prop: key,
     }));
-  }, [namespacedDynamicFactories, modelFactories, namespace]);
+  }, [modelFactories, namespace]);
 
   return (
-    <Firehose resources={resources}>
-      <ResourceDropdown
-        {...props}
-        placeholder={t('topology~Select an Application')}
-        dataSelector={['metadata', 'labels', 'app.kubernetes.io/part-of']}
-      />
-    </Firehose>
+    <DynamicWatchResourcesLoader
```

Comment:
you can move the `useResolvedExtension(isDynamicModelFactory)` call inside this component and resolve only the resources from the properties